fhir-persistence-jdbc/src/main/java/com/ibm/fhir/persistence/jdbc/dao/impl/FetchResourceChangesDAO.java

```diff
@@ -0,0 +1,134 @@
+/*
+ * (C) Copyright IBM Corp. 2021
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.ibm.fhir.persistence.jdbc.dao.impl;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Timestamp;
+import java.time.Instant;
+import java.util.ArrayList;
+import java.util.Calendar;
+import java.util.List;
+import java.util.TimeZone;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+import com.ibm.fhir.database.utils.api.IDatabaseTranslator;
+import com.ibm.fhir.persistence.ResourceChangeLogRecord;
+import com.ibm.fhir.persistence.ResourceChangeLogRecord.ChangeType;
+import com.ibm.fhir.persistence.exception.FHIRPersistenceException;
+
+/**
+ * Simple DAO to read records from the RESOURCE_CHANGE_LOG table
+ */
+public class FetchResourceChangesDAO {
+    private static final Logger logger = Logger.getLogger(FetchResourceChangesDAO.class.getName());
+    private static final Calendar UTC_CALENDAR = Calendar.getInstance(TimeZone.getTimeZone("UTC"));
+
+    private final IDatabaseTranslator translator;
+    private final String schemaName;
+    private final String resourceTypeName;
+    private final int resourceCount;
+    private final Long afterResourceId;
+    private final Instant fromTstamp;
+
+    /**
+     * Public constructor
+     * @param tx
+     * @param resourceCount
+     * @param resourceTypeName
+     * @param fromTstamp
+     * @param afterResourceId
+     */
+    public FetchResourceChangesDAO(IDatabaseTranslator tx, String schemaName, int resourceCount, String resourceTypeName, Instant fromTstamp, Long afterResourceId) {
+        this.translator = tx;
+        this.schemaName = schemaName;
+        this.resourceCount = resourceCount;
+        this.resourceTypeName = resourceTypeName;
+        this.fromTstamp = fromTstamp;
+        this.afterResourceId = afterResourceId;
+    }
+
+    /**
+     * Run the DAO command on the database connection
+     * @param c
+     * @return
+     * @throws FHIRPersistenceException
+     */
+    public List<ResourceChangeLogRecord> run(Connection c) throws FHIRPersistenceException {
+        ArrayList<ResourceChangeLogRecord> result = new ArrayList<>();
```

Comment:
any particular reason for. ArrayLIst? and not the generic?

Reply:
I changed it to List. I remember eclipse was glitching out on me here and I was trying to get rid of an error marker which wasn't actually an error. Ended up having to do a clean build.